src/main/java/seedu/address/model/score/Score.java

```diff
@@ -0,0 +1,109 @@
+package seedu.address.model.score;
+
+import static seedu.address.commons.util.CollectionUtil.requireAllNonNull;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+import java.util.Objects;
+
+import seedu.address.model.score.exceptions.BadDateException;
+
+/**
+ * Represents a Score in the address book.
+ * Guarantees: immutable; fields are validated; details are present and not null;
+ */
+public class Score {
+    // Identity field(s)
+    public final Title scoreName;
+    public final Integer scoreValue;
+    public final LocalDateTime scoreDate;
+
+    /**
+     * Constructs a {@code Score}.
+     *
+     * @param scoreName A valid Score name.
+     * @param scoreValue A valid Score value.
+     * @param scoreDate A valid Score date.
+     */
+    public Score(Title scoreName, Integer scoreValue, String scoreDate) {
+        requireAllNonNull(scoreName, scoreValue, scoreDate);
+
+        this.scoreName = scoreName;
+        this.scoreValue = scoreValue;
+
+        // Maybe can shift this validation when adding a score. TODO
+        try {
+            DateTimeFormatter formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm");
+            this.scoreDate = LocalDateTime.parse(scoreDate, formatter);
+        } catch (DateTimeParseException e) {
+            throw new BadDateException();
+        }
+    }
```

Comment:
Maybe we do not need the DateTime formate for the scoreDate since the date formate is enough.

Reply:
Sorry but can you try to explain it with more context? Use the code snippet ` character to show some emphasis.

Reply:
`DateTimeFormatter formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd")`
OR
instead of `LocalDateTime`, change to `LocalDate`